### ISSUE TITLE
Creating an option to skip the hardcoded ->where([primaryKey => $id]) in FindMethodTrait

### ISSUE THREAD

**Issue by Spriz** [FriendsOfCake/crud#628]
I faced some issues today when trying to use the `Crud.View` action on a model with a slug, even though I changed the finder with the correct attributes and everything, the `FindMethodTrait` keeps applying `->where(['Offers.id' => $id])` after my Finder is actually done.

Would it make sense to put this line: https://github.com/FriendsOfCake/crud/blob/master/src/Traits/FindMethodTrait.php#L60 into an `if` that can be bypassed with some configuration?

I know I could write my own Action and overwrite `_findRecord` - but it seems like a reasonable use case? What do you guys think?

**Comment by Spriz** [comment]
One can although circumvent things like this:

```php
public function view(string $slug): Response
    {
        /** @var \App\Model\Entity\Offer $offer */
        $offer = $this->Offers->find('BySlug', ['slug' => $slug])->firstOrFail();

        return $this->Crud->execute('view', $offer->id);
    }
```

**Comment by Spriz** [comment]
I am happy to throw in a PR I just wanted to know your opinion on this case before doing so :)

**Comment by ADmad** [comment]
If you do as below you wouldn't even need a custom finder :D

```
public function view(string $slug): Response
    {
        $this->Offers->setPrimaryKey('slug');

        return $this->Crud->execute();
    }
```

**Comment by Spriz** [comment]
Thanks @ADmad - This could have sideeffects for other code dependingn on `->getPrimaryKey()` during that request though :D

**Comment by ADmad** [comment]
Did you actually have any issue or you are just hypothsizing?

All that the `ViewAction` does is fetch a record and set view var. Even if customizing the finder like you propose is possible, why would someone do

```php
public function view(string $slug): Response
{
      $this->Crud->action()->setConfig('findMethod', ['BySlug' => ['slug' => $slug)]]);

        return $this->Crud->execute();
}
```
when the same can be achieved by doing

```php
public function view(string $slug): Response
{
    $this->set('offer', $this->Offers->find('BySlug', ['slug' => $slug]);
}
```

**Comment by Spriz** [comment]
@ADmad 

It's a hypothesis - we have quite some behaviors that use `$this->getPrimaryKey()` - and those would not be using `slug` field rather than the `id` if I understand the `$this->Offers->setPrimaryKey('slug')` correctly

Yeah, just doing that for view seems simpler, in our case it is for all the actions in the controller in said plugin/prefix where we want to use the `slug` over the `id` :) Maybe our usecase is too uncommon afterall.

TLDR: I would have liked to place `$this->Crud->action()->setConfig('findMethod', ['BySlug' => ['slug' => $slug)]]);` in `initialize` action :)
